tests/vaults.test.mjs

```diff
@@ -338,10 +339,81 @@ describe('Vaults', async () => {
       endingMemory = createVaultedTransferResult.Memory;
     });
 
+    it('should create a revokable vault for the recipient and the controller should be able to revoke that vault', async () => {
+      const controller = 'valid-controller-'.padEnd(43, 'a');
+      const { result: createVaultedTransferResult } =
+        await createVaultedTransfer({
+          quantity,
+          lockLengthMs,
+          recipient,
+          from: controller,
+          memory: sharedMemory,
+          revokable: true,
+        });
+
+      const vaultId = createVaultedTransferResult.Messages[0].Tags.find(
+        (tag) => tag.name === 'Vault-Id',
+      ).value;
+
+      // ensure vault id is defined
+      assert.ok(vaultId);
+
+      const expectedVaultData = {
+        balance: quantity,
+        controller,
+        startTimestamp: STUB_TIMESTAMP,
+        endTimestamp: STUB_TIMESTAMP + lockLengthMs,
+      };
+
+      const createdVaultData = await assertVaultExists({
+        vaultId,
+        address: recipient,
+        memory: createVaultedTransferResult.Memory,
+      });
+      assert.deepEqual(createdVaultData, expectedVaultData);
+
+      // Revoke the vault
+      const result = await handle({
+        options: {
+          Tags: [
+            { name: 'Action', value: 'Revoke-Vault' },
+            { name: 'Recipient', value: recipient },
+            { name: 'Vault-Id', value: vaultId },
+          ],
+          From: controller,
+          Owner: controller,
+        },
+        memory: createVaultedTransferResult.Memory,
+      });
+
+      assert.deepEqual(result.Messages.length, 2);
+      const recipientMessageAfterRevoke = result.Messages.find((msg) =>
+        msg.Tags.find(
+          (tag) => tag.name === 'Action' && tag.value === 'Revoke-Vault-Notice',
+        ),
+      );
+      assert.ok(recipientMessageAfterRevoke);
+      const recipientVaultDataAfterRevoke = JSON.parse(
+        recipientMessageAfterRevoke.Data,
+      );
+      assert.deepEqual(recipientVaultDataAfterRevoke, expectedVaultData);
+
+      const controllerMessageAfterRevoke = result.Messages.find((msg) =>
+        msg.Tags.find(
+          (tag) => tag.name === 'Action' && tag.value === 'Revoke-Vault-Notice',
+        ),
+      );
+      assert.ok(controllerMessageAfterRevoke);
+      const controllerVaultDataAfterRevoke = JSON.parse(
+        controllerMessageAfterRevoke.Data,
+      );
+      assert.deepEqual(controllerVaultDataAfterRevoke, expectedVaultData);
+
+      endingMemory = result.Memory;
```

Comment:
This test needs to check the balance of the revoker for increase. If here it would have caught the logical error noted above.

Reply:
you could/should also check for a credit-notice sent to sender.

Reply:
881b3c6